src/System.Private.CoreLib/shared/System/Globalization/CompareInfo.Unix.cs

```diff
@@ -517,111 +511,12 @@ private unsafe bool StartsWith(ReadOnlySpan<char> source, ReadOnlySpan<char> pre
             Debug.Assert(!prefix.IsEmpty);
             Debug.Assert((options & (CompareOptions.Ordinal | CompareOptions.OrdinalIgnoreCase)) == 0);
 
-            if (_isAsciiEqualityOrdinal && CanUseAsciiOrdinalForOptions(options))
-            {
-                if (source.Length < prefix.Length)
-                {
-                    return false;
-                }
-
-                if ((options & CompareOptions.IgnoreCase) == CompareOptions.IgnoreCase)
-                {
-                    return StartsWithOrdinalIgnoreCaseHelper(source, prefix, options);
-                }
-                else
-                {
-                    return StartsWithOrdinalHelper(source, prefix, options);
-                }
-            }
-            else
-            {
-                fixed (char* pSource = &MemoryMarshal.GetReference(source))
-                fixed (char* pPrefix = &MemoryMarshal.GetReference(prefix))
-                {
-                    return Interop.Globalization.StartsWith(_sortHandle, pPrefix, prefix.Length, pSource, source.Length, options);
-                }
-            }
-        }
-
-        private unsafe bool StartsWithOrdinalIgnoreCaseHelper(ReadOnlySpan<char> source, ReadOnlySpan<char> prefix, CompareOptions options)
-        {
-            Debug.Assert(!GlobalizationMode.Invariant);
-
-            Debug.Assert(!source.IsEmpty);
-            Debug.Assert(!prefix.IsEmpty);
-            Debug.Assert(_isAsciiEqualityOrdinal);
-            Debug.Assert(source.Length >= prefix.Length);
-
-            int length = prefix.Length;
-
-            fixed (char* ap = &MemoryMarshal.GetReference(source))
-            fixed (char* bp = &MemoryMarshal.GetReference(prefix))
+            if (source.Length < prefix.Length)
             {
-                char* a = ap;
-                char* b = bp;
-
-                while (length != 0 && (*a < 0x80) && (*b < 0x80) && (!s_highCharTable[*a]) && (!s_highCharTable[*b]))
-                {
-                    int charA = *a;
-                    int charB = *b;
-
-                    if (charA == charB)
-                    {
-                        a++; b++;
-                        length--;
-                        continue;
-                    }
-
-                    // uppercase both chars - notice that we need just one compare per char
-                    if ((uint)(charA - 'a') <= (uint)('z' - 'a')) charA -= 0x20;
-                    if ((uint)(charB - 'a') <= (uint)('z' - 'a')) charB -= 0x20;
-
-                    if (charA != charB)
-                        return false;
-
-                    // Next char
-                    a++; b++;
-                    length--;
-                }
-
-                if (length == 0) return true;
-                return Interop.Globalization.StartsWith(_sortHandle, b, length, a, length, options);
+                return false;
```

Comment:
I don't think you can make this assumption in the non-ordinal case.

```
            Thread.CurrentThread.CurrentCulture = CultureInfo.GetCultureInfo("fr-FR");

            string str1 = "œ";
            string str2 = "oe";

            Console.WriteLine(str1.StartsWith(str2, StringComparison.CurrentCulture));
```
This will print true even though str2.Length > str1.Length

Reply:
let's say that this was not my best idea ;p